WordPress/Classes/ViewRelated/Stats/Insights/Insights Management/AddInsightTableViewController.swift

```diff
@@ -39,19 +63,36 @@ class AddInsightTableViewController: UITableViewController {
         reloadViewModel()
         WPStyleGuide.configureColors(view: view, tableView: tableView)
         WPStyleGuide.configureAutomaticHeightRows(for: tableView)
-        tableView.accessibilityIdentifier = "Add Insight Table"
+        tableView.accessibilityIdentifier = TextContent.title
+
+        if FeatureFlag.statsNewAppearance.enabled {
+            tableView.isEditing = true
+            tableView.allowsSelectionDuringEditing = true
+        }
 
         navigationItem.leftBarButtonItem = UIBarButtonItem(image: .gridicon(.cross), style: .plain, target: self, action: #selector(doneTapped))
     }
 
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
 
-        if selectedStat == nil {
+        // Catches user swiping down to dismiss
```

Comment:
I was wondering if we could do this a better way as I'm not a fan of the 3 if checks in a row.  It occurred to me that for dismissing perhaps you could try UIAdaptivePresentationControllerDelegate presentationControllerDidDismiss or perhaps there is a different reason that it cannot be used?